### PR TITLE
[NetCDF] rebuild against new HDF5 1.12.0 build

### PR DESCRIPTION
**Pull request by visr**
I did #1112, #2111 did I?

### CODE REVIEW

File: N/NetCDF/build_tarballs.jl [hunk 1]
```diff
@@ -87,9 +86,14 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("HDF5_jll"),
+    Dependency("HDF5_jll", v"1.12.0"),
```
Comment (giordano) [inline N/NetCDF/build_tarballs.jl:89]: Note that we have two ways of expressing compatibilities:
```julia
Dependency("Foo_jll", v"1.2.3")
```
means "build with `Foo_jll` v1.2.3, but it's compatible with any version, while (note: I don't exactly remember the syntax by heart)
```julia
Dependency(PackageSpec(; name = "Foo_jll", version = "1.2.3"))
```
means "this is compatible with Foo_jll v1.2.3"

**Comment by visr** [comment]
This should be good to merge now.

**Comment by giordano** [comment]
Problems with libmbedtls: https://github.com/JuliaRegistries/General/pull/24872/checks?check_run_id=1418037742

**Comment by visr** [comment]
Yeah bummer, I tried looking around a bit locally on Windows but got this error, probably similar.

```
julia> using NetCDF_jll
ERROR: InitError: could not load library "C:\Users\visser_mn\.julia\artifacts\bde5472e8aa0512b49dde365ece4ead2a6cb2885\bin\libnetcdf-18.dll"
The specified procedure could not be found.
Stacktrace:
 [1] dlopen(::String, ::UInt32; throw_error::Bool) at C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.5\Libdl\src\Libdl.jl:109
 [2] dlopen at C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.5\Libdl\src\Libdl.jl:109 [inlined] (repeats 2 times)
 [3] __init__() at C:\Users\visser_mn\.julia\packages\NetCDF_jll\KYp8V\src\wrappers\x86_64-w64-mingw32.jl:46
 [4] _include_from_serialized(::String, ::Array{Any,1}) at .\loading.jl:697
 [5] _require_search_from_serialized(::Base.PkgId, ::String) at .\loading.jl:782
 [6] _require(::Base.PkgId) at .\loading.jl:1007
 [7] require(::Base.PkgId) at .\loading.jl:928
 [8] require(::Module, ::Symbol) at .\loading.jl:923
during initialization of module NetCDF_jll
```

Although when checking with `using DependencyWalker; Library(NetCDF_jll.libnetcdf_path)` it could find the mbedtls library without issues.

**Comment by giordano** [comment]
@visr what version of `NetCDF_jll` is that?

**Comment by visr** [comment]
That is the master version, so after this PR merged.

**Comment by giordano** [comment]
So you installed `NetCDF_jll#master`?

**Comment by giordano** [comment]
See https://github.com/JuliaPackaging/Yggdrasil/blob/e10ffd25f8b5030da5d768588257ba5facc5d5da/A/AzStorage/build_tarballs.jl#L45-L51 We need to fix also the version we use for building MbedTLS

**Pull request merged**